docs/content/guide/component.ngdoc

```diff
@@ -0,0 +1,453 @@
+
+### Comparison between Directive definition and Component definition
+
+|                   | Directive            | Component       |
+|-------------------|----------------------|-----------------|
+| bindings          | No                   | Yes (binds to controller)      |
+| bindToController  | Yes (default: false) | No (use bindings instead)       |
+| compile function  | Yes                  | No       |
+| controller        | Yes                  | Yes      |
+| controllerAs      | Yes (default: false) | Yes (default: `$ctrl`)       |
+| link functions    | Yes                  | No       |
+| multiElement      | Yes                  | No       |
+| priority          | Yes                  | No       |
+| require           | Yes                  | Yes      |
+| restrict          | Yes                  | No (restricted to elements only)      |
+| scope             | Yes (default: false) | No (scope is always isolate)      |
+| template          | Yes                  | Yes, injectable      |
+| templateNamespace | Yes                  | No       |
+| templateUrl       | Yes                  | Yes, injectable       |
+| terminal          | Yes                  | No       |
+| transclude        | Yes (default: false) | Yes (default: false)      |
+
+
+## Component-based application architecture
+
+As already mentioned, the component helper makes it easier to create your application with
+a component-based application architecture. But what makes a component beyond the options that
+the component helper has?
+
+- Components only control their own View and Data
+Components should never modify any data or DOM that is out of their own scope. Normally, in Angular
+it is possible to modify data anywhere in the application through scope inheritance and watches. This
+is practical, but can also lead to problems when it is not clear which part of the application is
+responsible for modifying the data. That is why component directives are isolate scope, so a whole
+class of scope manipulation is not possible.
+
+- Components have a well-defined public API - Inputs and Outputs
+However, scope isolation only goes so far, because Angular uses two-way binding. So if you pass
+an object to a component like this - `bindings: {item: '='}`, and modify one of its properties, the
+change will be reflected in the parent component. For components however, only the component that owns
+the data should modify it, to make it easy to reason what data is changed when.
+
+For that reason, components should follow a few simple conventions:
+
+  - Inputs are realized with `@` and `=` bindings
+    ```js
+      bindings: {
+        hero: `=`,
+      }
+    ```
+  - Output is realized with `&` bindings, which function as callbacks to component events
+    ```js
+      bindings: {
+        onDelete: '&',
+        onUpdate: '&'
+      }
+    ```
+  - Instead of manipulating Input Data, the component calls the Output Event with the changed data
```

Comment:
So what is `hero` then if not the input data ? Do you suggest components should create copies of their inputs ?
It's not very clear what this guidelines means (and I am not sure I agree :stuck_out_tongue:).